fmf_scheduler.py

```diff
@@ -0,0 +1,188 @@
+        node.sources[-1]), node.data["test"])
+    modulename = os.path.basename(
+        node.sources[-1]).split(".", 1)[0]
+    # in case of referencing  use original data tree for info
+    if "@" in node.name and not os.path.exists(modulepath):
+        modulepath = os.path.join(os.path.dirname(
+            node.sources[-2]), node.data["test"])
+        modulename = os.path.basename(
+            node.sources[-2]).split(".", 1)[0]
+    test_func = make_check_function(target=target)
+    logger.debug("Try to import: %s from path %s", modulename, modulepath)
+    moduleimp = imp.load_source(modulename, modulepath)
+    inernalclass = getattr(moduleimp, second_class_name)
+    # more verbose output
+    # full_class_name = '{0}_{1}'.format(first_class_name, second_class_name)
+    full_class_name = '{0}'.format(first_class_name)
+    return type(full_class_name, (DynamicClassBase,), {'test': test_func,
+                                                       'backendclass': inernalclass
+                                                       })
+
+def nosetests_class_fmf_generator(fmfpath, target_name, log_level, ruleset_tree_path=None, filter_names=None, filters=None):
+    """
+    generates dynamic test classes for nosetest or unittest scheduler based on FMF metadata.
+
+    :param fmfpath: path to checks
+    :param target_name: what is the target object
+    :param log_level:
+    :param ruleset_tree_path:
+    :return:
+    """
+    target = Target(target_name, log_level)
+    test_classes = {}
+    if not ruleset_tree_path:
+        ruleset_tree_path = fmfpath
+    ruleset_metadatatree = ExtendedTree(ruleset_tree_path)
+    metadatatree = ExtendedTree(fmfpath)
+    ruleset_metadatatree.references(metadatatree)
+    for node in ruleset_metadatatree.prune(names=filter_names, filters=filters):
+        if node.data.get("class") or node.data.get("test"):
+            logger.debug("node (%s) contains test and class item", node.name)
+
+            if node.data.get("class") in CLASS_MAPPING:
+                logger.debug("Using pure FMF metadata for %s (class %s)", node.name, node.data.get("class"))
+                test_class = make_base_fmf_class_abstract(node=node, target=target)
+            else:
+                logger.debug("searching for %s", node.name)
+                test_class = make_wrapped_fmf_class(node=node, target=target)
+            if is_compatible(target_type=target.target_type, check_instance=test_class.backendclass()):
+                test_classes[test_class.__name__] = test_class
+                logger.debug("Test added: %s", node.name)
+            else:
+                logger.debug("Test (not target): %s", node.name)
+        else:
+            if "__pycache__" not in node.name:
+                logger.warning("error in fmf config for node (missing test and class items): %s (data: %s) ", node.name, node.data)
+    return test_classes
+
+
+def scheduler_opts(target_name=None, checks=None, ruleset_path=None,
```

Comment:
colin functionality has to remain as it is. scheduler_opts has to be valid only for FMF.

Reply:
sure, thats why it is directly in FMF scheduler and not inside colin, but finally we, remove colin and use this and make it backward complatible, so finally there will be no changes for current colin users